### PR TITLE
Add Cardano Blocks APIs, Implement Tests, and Refactor Code

### PR DESCRIPTION
**Pull request by hieu-dd**
# Pull Request (PR) Summary

- This PR introduces new APIs for Cardano blocks, along with unit and integration tests. It also includes code refactoring and bug fixes.

## Changes Description

In this PR, I have added new APIs to the SDK to handle interactions with Cardano blocks. These APIs allow users to retrieve information about specific blocks, list blocks within certain parameters, and track block changes over time.

To ensure the functionality and reliability of these new APIs, I have implemented both unit tests and integration tests. These tests cover a wide range of scenarios to ensure that the APIs handle expected and edge cases effectively.

Additionally, I have conducted a thorough refactoring of the existing codebase. This refactoring aims to improve the readability, performance, and maintainability of the code. During the refactoring process, several bugs were identified and fixed, enhancing the overall stability and functionality of the SDK.

## Related Issue(s)

- Fixes # (issue number)  // Please replace with actual issue number if applicable

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Other (please describe)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] My PR is targeting the `main` branch
- [x] I have added necessary documentation (if appropriate)
- [x] I have added tests that prove my fix is effective or my feature works
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/blocks/model/BlockContent.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * @property blockVrf VRF key of the block
+ * @property previousBlock Hash of the previous block
+ * @property nextBlock Hash of the next block
+ * @property confirmations Number of block confirmations
+ */
+
+@Serializable
+data class BlockContent(
+    @SerialName("time") val time: Int,
+    @SerialName("height") val height: Int?,
```
Comment (HoangNguyen219) [inline core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/blocks/model/BlockContent.kt]: Mấy field nullable này không set default là null thì có lỗi `Serialization error` khi nó null không nhỉ

File: core/src/commonTest/kotlin/dev/kryptonreborn/blockfrost/integrationtest/CardanoAddressApiIntegrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+class CardanoAddressApiIntegrationTest : BaseIntegrationTest() {
+    private val address =
+        "addr1qxqs59lphg8g6qndelq8xwqn60ag3aeyfcp33c2kdp46a09re5df3pzwwmyq946axfcejy5n4x0y99wqpgtp2gd0k09qsgy6pz"
+    private val asset = "00000002df633853f6a47465c9496721d2d5b1291b8398016c0e87ae6e7574636f696e"
+
+    @Test
+    fun testGetAddress() =
+        runIntegrationTest {
+            val result = blockfrostClient.getSpecificAddress(address)
+            println(result.getOrNull())
```
Comment (HoangNguyen219) [inline core/src/commonTest/kotlin/dev/kryptonreborn/blockfrost/integrationtest/CardanoAddressApiIntegrationTest.kt]: xóa hết mấy chỗ println nhé ạ

File: core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/blocks/model/BlockContent.kt (hunk 1 above)
Comment (HoangNguyen219) [reply core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/blocks/model/BlockContent.kt]: Với lại @SerialName giống tên thuộc tính thì thêm làm gì ạ

Comment (hieu-dd) [reply core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/blocks/model/BlockContent.kt]: > Mấy field nullable này không set default là null thì có lỗi `Serialization error` khi nó null không nhỉ

ko lỗi đâu e

**Review by iTanChi — APPROVED** [review]
